### ISSUE TITLE
[2020] DivideMix: Learning with Noisy Labels as Semi-supervised Learning

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#120]
## 0. 論文

| タイトル | DivideMix: Learning with Noisy Labels as Semi-supervised Learning |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2002.07394 |
| 著者    | Junnan Li, Richard Socher, Steven C.H. Hoi (Salesforce Research) |
| 投稿日付 | ICLR 2020 |

## 1. どんなもの？

Noisy label を含むデータセットによるモデルの学習において、学習時のロスからデータセットを clean label data と noisy label data の2つに分類し、clean label は通常の学習を、noisy label は元のラベルを捨てて semi-supervised な手法で学習を行うことでロバストなモデルを作成する手法である DivideMix を提案。
学習時のロスを2つの GMM に fitting させることでデータセットを分割するが、1つのモデルで行うと confirmation bias が発生してしまうため、モデルを2つ用意して互いが使用するデータセットをもう一方のモデルの出力で判断する、という方法で回避している。

![スクリーンショット 2021-02-20 19 21 42](https://user-images.githubusercontent.com/1255485/108594066-10087380-73bb-11eb-8b67-3debdc64b6d8.png)

semi-supervised の部分は MixMatch を DivideMix 向けに改善しており、2つのモデルで clean label の refinement や noisy label の推論を行っている。
CIFAR10/100 によるシミュレーションデータ、Clothing1M, WebVision での実験で既存手法のスコアを大きく更新。

## 2. 先行研究と比べてどこがすごい？

- noisy label の学習において、 データセットを動的に変えながら semi-supervised を組み合わせた学習方法を提案
- confirmation bias を避けるため、2つのモデルを同時に学習し、お互いのデータセットを分割する
- 既存手法のスコアを大きく改善

## 3. 技術や手法のキモはどこ？

モデル学習時の各サンプルのロスを clean, noisy 2つのコンポーネントを持つ GMM でモデリングし、これにより学習データを clean set と noisy set の2つに分け、毎 epoch でこのデータ分割を更新しながら、clean set は通常通り学習、noisy set はラベルを捨てて MixMatch で学習する。
また学習の際は、確証バイアスを避けるため2つのモデルを学習し、片方のモデルのロスにより分割したデータをもう一方のモデルの学習に用いる Co-Divide を行っている。
細かい流れは下記。

- 学習初期は warmup として普通の Cross Entropy で全データで学習を行う
  - asymmetric noise の場合、overfit を避けるためにエントロピーを正則化項として追加して学習する
- warmup 後、DivideMix を適用する
  - データを labeled (clean) と unlabeled (noisy) に分割する
  - M回異なる augmentation を入れてモデルによる推論を行い、その結果からデータのラベルを更新
    - labeled は元のラベルとM回の推論結果の重み付き和を sharpen したもの
    - unlabeled はM回 * 2つのモデルの推論の平均
  - 更新したラベルをもとに分割されたデータセットで MixMatch でモデルを学習

## 4. どうやって有効だと検証した？

CIFAR10/100 によるシミュレーションデータ、Clothing1M, WebVision での実験で既存手法のスコアを大きく更新した。

![スクリーンショット 2021-02-20 19 21 51](https://user-images.githubusercontent.com/1255485/108594069-139bfa80-73bb-11eb-9c79-f1bb1b21a974.png)

## 5. 議論はある？

noisy label の問題に semi-supervised の他の手法を適用したり、他のドメインに DivideMix を適用したりできそうとのこと。

## 6. 次に読むべき論文は？

- MixMatch: A Holistic Approach to Semi-Supervised Learning
    - https://arxiv.org/abs/1905.02249
    - David Berthelot, Nicholas Carlini, Ian Goodfellow, Nicolas Papernot, Avital Oliver, Colin Raffel (Google research)
    - 半教師あり学習の手法
